### ISSUE TITLE
[BUG] webgl backend results in dashed lines

### ISSUE THREAD

**Issue by vroger11** [bokeh/bokeh#12058]
### Software versions

Python version      :  3.8.12 (default, Oct 12 2021, 13:49:34) 
IPython version     :  8.1.1
Tornado version     :  6.1
Bokeh version       :  2.4.2
BokehJS static path :  /home/vroger/.miniconda3/envs/data_analysis/lib/python3.8/site-packages/bokeh/server/static
node.js version     :  v17.7.1
npm version         :  8.5.4
Operating system    :  Linux-5.16.14-1-MANJARO-x86_64-with-glibc2.17

### Browser name and version

Visual studio code: 1.65.2

### Jupyter notebook / Jupyter Lab version

jupyter notebook: 6.4.8

### Expected behavior

The result should be the same using webgl or not using it.

### Observed behavior

Using webgl may result in showing dashes instead of lines.

### Example code

```Python
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
import pandas as pd

output_notebook()

df = pd.DataFrame({'x':7}, index=pd.RangeIndex(20))
df['y'] = df.index.values
df.loc[10:15,'y'] = 10

p = figure(output_backend='webgl')
p.line(df.index, df.x)
p.line(df.index, df.y)
show(p)
```

### Screenshots

![Screenshot_20220323_170814](https://user-images.githubusercontent.com/6843012/159744575-ea5a4e21-8fd3-4be8-a62c-7caf0e6c983d.png)

Dashes appear when the line is horizontal (or a portion of the line).
I do not observe this behavior without the webgl backend.

**Comment by bryevdv** [comment]
WebGL has undergone extensive re-work with lots of new tests on `branch-3.0`. There is a good chance this is already resolved there. cc @ianthomas23

**Comment by ianthomas23** [comment]
This is definitely a bug that still exists on `branch-3.0`. Here is a simpler reproducer:
```python
from bokeh.plotting import figure, show
p = figure(width=200, height=200, output_backend='webgl')
p.grid.visible = False
p.line([0, 0.5, 1], [1, 1, 1], line_width=10)
show(p)
```
![line](https://user-images.githubusercontent.com/580326/159763504-84ea87fa-ff66-45d3-9969-b1809ed3ba5f.png)

It only happens for `line_join="bevel"`, which is unfortunately the default. As a workaround, using
```python
p.line([0, 0.5, 1], [1, 1, 1], line_width=10, line_join="miter")
```
or `line_join="round"` gives you what you want to see.

I know what the problem is, so I should have a fix for the 3.0 release.

Thanks for the bug report @vroger11.

**Comment by ianthomas23** [comment]
Closed by #12065
